var/spack/repos/builtin/packages/m4/package.py

```diff
@@ -39,6 +39,16 @@ class M4(Package):
 
     depends_on('libsigsegv', when='+sigsegv')
 
+    def _make_executable(self, name):
+        return Executable(join_path(self.prefix.bin, name))
+
+    def setup_dependent_package(self, module, dependent_spec):
+        # m4 is very likely to be a build dependency, so we add the
```

Comment:
The issues is that package `X` has a build dependency on `autoconf`, and `autoconf` has a run dependency on `m4`. This means that `X` has, at build time, no access to `m4`.

I believe the confusion is that `m4` is (as declared) a run-time dependency from autoconf's point of view, but is -- by transitivity -- also a build-time dependency from `X`'s point of view. The latter is not handled by Spack.